pandas/core/indexing.py

```diff
@@ -1115,9 +1115,12 @@ def _validate_key(self, key, axis: Axis):
         # slice of labels (where start-end in labels)
         # slice of integers (only if in the labels)
         # boolean not in slice and with boolean index
+        ax = self.obj._get_axis(axis)
         if isinstance(key, bool) and not (
-            is_bool_dtype(self.obj._get_axis(axis))
-            or self.obj._get_axis(axis).dtype.name == "boolean"
+            is_bool_dtype(ax)
+            or ax.dtype.name == "boolean"
+            or isinstance(ax, MultiIndex)
+            and is_bool_dtype(ax.get_level_values(0))
```

Comment:
So if one wanted to index the bools in the second level (and the first level was not boolean), does that take a different code path?

Reply:
You mean something like ``df.loc[(slice(None), True)]``?

This runs through a different path, same reason why ``df.loc[(True, )]`` worked already.